### PR TITLE
Add Vanilla SwiftUI example

### PR DESCRIPTION
**Pull request by mltbnz**
Adds a simple example how to use Composable Navigator with Vanilla SwiftUI.

### CODE REVIEW

File: Example/Vanilla Example/Vanilla Example/ContentView.swift [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import SwiftUI
+import ComposableNavigator
+
+struct HomeScreen: Screen {
+    let presentationStyle: ScreenPresentationStyle = .push
```
Comment (oliverlist) [inline Example/Vanilla Example/Vanilla Example/ContentView.swift:5]: this file still uses 4 spaces

File: Example/Vanilla SwiftUI Example/Vanilla SwiftUI Example/CapacityView.swift [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import ComposableNavigator
+import SwiftUI
+
+struct CapacityScreen: Screen {
+  let capacity: Int64
```
Comment (ohitsdaniel) [inline Example/Vanilla SwiftUI Example/Vanilla SwiftUI Example/CapacityView.swift:5]: Can we change to Int here? Might confusing for beginners.

File: Example/Vanilla SwiftUI Example/Vanilla SwiftUI Example/HomeView.swift [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  let presentationStyle: ScreenPresentationStyle = .push
+  
+  struct Builder: NavigationTree {
+    var builder: some PathBuilder {
+      Screen(
+        HomeScreen.self,
+        content: { HomeView() },
+        nesting: {
+          DetailScreen.Builder()
+        })
```
Comment (ohitsdaniel) [inline Example/Vanilla SwiftUI Example/Vanilla SwiftUI Example/HomeView.swift:14]: Missing newline :)

**Review by ohitsdaniel — APPROVED** [review]
Just two tiny comments :)

**Pull request merged**